src/main/java/org/openrewrite/java/template/internal/TemplateCode.java

```diff
@@ -64,12 +64,11 @@ public static <T extends JCTree> String process(T tree, List<JCTree.JCVariableDe
                 jarNames.addAll(ClasspathJarNameDetector.classpathFor(parameter, ImportDetector.imports(parameter)));
             }
             if (!jarNames.isEmpty()) {
-                // It might be preferable to enumerate exactly the needed dependencies rather than the full classpath
-                // But this is expedient
-                // See https://github.com/openrewrite/rewrite-templating/issues/86
-                // String classpath = jarNames.stream().map(jarName -> '"' + jarName + '"').sorted().collect(joining(", "));
-                // builder.append("\n    .javaParser(JavaParser.fromJavaVersion().classpath(").append(classpath).append("))");
-                builder.append("\n    .javaParser(JavaParser.fromJavaVersion().classpath(JavaParser.runtimeClasspath()))");
+                String joinedJarNames = jarNames.stream().collect(joining(", ", "\"", "\""));
+                builder.append("\n    .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, ")
+                        .append(joinedJarNames)
+                        .append("))\n    ");
```

Comment:
Hmm; this now breaks the tests we have for the TemplateProcessor, as the `ctx` can not be found there, and it looks like that might not be easy to introduce where called from openrewrite/rewrite.
https://github.com/openrewrite/rewrite-templating/blob/2c9dcfd3f3186851d871b16531e63430dbc2fb6f/src/test/resources/template/ParameterReuseRecipe.java#L23-L28

Luckily it's not a feature we use often, but still it'd be a shame to break that here.

Reply:
We do use it in rewrite-micrometer, IIRC.

Reply:
There's indeed three places:
1. https://github.com/openrewrite/rewrite-apache/blob/df33d276534aaaea4c2f2f93ad681954084e09b2/src/main/java/org/openrewrite/apache/commons/lang/IsNotEmptyToJdk.java#L74-L77
2. https://github.com/openrewrite/rewrite-migrate-java/blob/6b98403236ad24c7d949750cd1c7af5fb94eebd8/src/main/java/org/openrewrite/java/migrate/UseJavaUtilBase64.java#L126-L127
3. https://github.com/openrewrite/rewrite-micrometer/blob/277c471b5d22db127cd86f00eff7e8ac933c8c9b/src/main/java/org/openrewrite/micrometer/misk/MigrateEmptyLabelMiskCounter.java#L71-L73

Reply:
I've deferred changing things here for now with a new condtional: https://github.com/openrewrite/rewrite-templating/pull/139/commits/b44be876bf5f62c4dbc5d378d797505ebc9f4170

We _could_ change the `Semantics` class and `PattenrBuilder` to pass along an additional `ExecutionContext` argument, but I figured we need not go through that trouble just yet for those rare cases.

https://github.com/openrewrite/rewrite-templating/blob/022373a047323b6a401d13adc986284f290363ea/src/main/java/org/openrewrite/java/template/Semantics.java#L24-L34